Assets/HoloToolkit/Input/Scripts/Microphone/MicStream.cs

```diff
@@ -0,0 +1,174 @@
+		/// <returns>error code or 0</returns>
+		[DllImport("MicStreamSelector", ExactSpelling = true)]    public static extern int MicInitializeCustomRate(int category, int samplerate);
+
+		/// <summary>
+		/// Call this to start receiving data from a microphone. Then, each frame, call MicGetFrame.
+		/// </summary>
+		/// <param name="keepData">If true, all data will stay in the queue, if the client code is running behind. This can lead to significant audio lag, so is not appropriate for low-latency situations like reall-time voice chat.</param>
+		/// <param name="previewOnDevice">If true, the audio from the microphone will be played through your speakers.</param>
+		/// <param name="micsignal">Optional (can be null): This callback will be called when data is ready for MicGetFrame</param>
+		/// <returns>error code or 0</returns>
+		[DllImport("MicStreamSelector")]    public static extern int MicStartStream(bool keepData, bool previewOnDevice, LiveMicCallback micsignal);
+
+		/// <summary>
+		/// Call this to start receiving data from a microphone. Then, each frame, call MicGetFrame.
+		/// </summary>
+		/// <param name="keepData">If true, all data will stay in the queue, if the client code is running behind. This can lead to significant audio lag, so is not appropriate for low-latency situations like reall-time voice chat.</param>
+		/// <param name="previewOnDevice">If true, the audio from the microphone will be played through your speakers.</param>
+		/// <returns>error code or 0</returns>
+		public static int MicStartStream(bool keepData, bool previewOnDevice)
+		{
+			return MicStartStream(keepData, previewOnDevice, null);
+		}
+
+		/// <summary>
+		/// Shuts down the connection to the microphone. Data will not longer be received from the microphone.
+		/// </summary>
+		/// <returns>error code or 0</returns>
+		[DllImport("MicStreamSelector")]    public static extern int MicStopStream();
+
+		/// <summary>
+		/// Begins recording microphone data to the specified file.
+		/// </summary>
+		/// <param name="filename">The file will be saved to this name. Specify only the wav file's name with extensions, aka "myfile.wav", not full path</param>
+		/// <param name="previewOnDevice">If true, will play micstream in speakers</param>
+		/// <returns></returns>
+		[DllImport("MicStreamSelector")]    public static extern int MicStartRecording(string filename, bool previewOnDevice);
+
+		/// <summary>
+		/// Finishes writing the file recording started with MicStartRecording.
+		/// </summary>
+		/// <param name="sb">returns the full path to the recorded audio file</param>
+        [DllImport("MicStreamSelector")]    public static extern void MicStopRecording(StringBuilder sb);
+
+		/// <summary>
+		/// Finishes writing the file recording started with MicStartRecording.
+		/// </summary>
+		/// <returns>the full path to the recorded audio file</returns>
+		public static string MicStopRecording()
+		{
+			StringBuilder builder = new StringBuilder(1024);
```

Comment:
Any specific reason for 1024 and not just new SB( )?
Please make it a const with a PascalCased name.